Library/Homebrew/os/mac/keg.rb

```diff
@@ -0,0 +1,102 @@
+require "os/mac/pathname"
+
+module InstallNameToolExtention
+  def change_dylib_id(id, file)
+    puts "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}" if ARGV.debug?
+    install_name_tool("-id", id, file)
+  end
+
+  def change_install_name(old, new, file)
+    puts "Changing install name in #{file}\n  from #{old}\n    to #{new}" if ARGV.debug?
+    install_name_tool("-change", old, new, file)
+  end
+
+  def install_name_tool(*args)
+    @require_install_name_tool = true
+    tool = MacOS.install_name_tool
+    system(tool, *args) || raise(ErrorDuringExecution.new(tool, args))
+  end
+
+  def require_install_name_tool?
+    !!@require_install_name_tool
+  end
+end
+
+
+class Keg
```

Comment:
I share your feeling. I wasn't happy about `keg_relocate` as well. I will see what I can do to improve it.